pkg/driver/driver_azure.go

```diff
@@ -158,6 +160,30 @@ func (d *AzureDriver) getVMParameters(vmName string, networkInterfaceReferenceID
 	return VMParameters
 }
 
+func isUrnProvided(d *AzureDriver) bool {
+	return d.AzureMachineClass.Spec.Properties.StorageProfile.ImageReference.URN != nil &&
+		*d.AzureMachineClass.Spec.Properties.StorageProfile.ImageReference.URN != ""
+}
+
+func getAzureImageDetails(d *AzureDriver) (publisher, offer, sku, version string) {
+	if isUrnProvided(d) {
+		splits := strings.Split(*d.AzureMachineClass.Spec.Properties.StorageProfile.ImageReference.URN, ":")
+		if len(splits) != 4 {
```

Comment:
Minor: this check shouldn't be needed, as this is taken care of as part of validation.

Reply:
removed